### PR TITLE
feat: disable kbcli addons in batch

### PR DESCRIPTION
**Pull request by SuperAayush**
The PR contains the code to make the kbcli to disable more than one addon at once. I have not tested the code yet since I am not aware of how to setup kbcli locally for testing purposes and have tried reaching out to @shanshanying for the same.

I respect the busy schedule all the maintainers but would be helpful if I can get KT on how should once setup kbcli locally mainly for testing purposes so that I can make the PR for this issue as well: #278 

I am aware that this PR still needs many changes hence I am opening this as a draft PR.

Fixes: #237

### CODE REVIEW

**Comment by shanshanying** [comment]
build with
```bash
make kbcli
```
and get follwing err msg
image

To test, just run
```bash
make test
```

Test cases for addons should be added to `pkg/cmd/addon/addon_test.go` .

File: pkg/cmd/addon/addon.go [hunk 1]
```diff
@@ -266,15 +266,17 @@ func newDisableCmd(f cmdutil.Factory, streams genericiooptions.IOStreams) *cobra
 	cmd := &cobra.Command{
 		Use:               "disable ADDON_NAME",
 		Short:             "Disable an addon.",
-		Args:              cobra.ExactArgs(1),
+		Args:              cobra.MinimumNArgs(1), //This will accept more than one argument rather than keeping the args to exact
```
Comment (shanshanying) [inline pkg/cmd/addon/addon.go:269]: add an extra space btw '//' and comment.

File: pkg/cmd/addon/addon.go [hunk 2]
```diff
@@ -266,15 +266,17 @@ func newDisableCmd(f cmdutil.Factory, streams genericiooptions.IOStreams) *cobra
...
 		ValidArgsFunction: util.ResourceNameCompletionFunc(f, types.AddonGVR()),
 		Run: func(cmd *cobra.Command, args []string) {
-			util.CheckErr(o.init(args))
-			util.CheckErr(o.fetchAddonObj())
-			util.CheckErr(o.checkBeforeDisable())
-			util.CheckErr(o.complete(o, cmd, args))
-			util.CheckErr(o.CmdComplete(cmd))
-			util.CheckErr(o.Run())
+			for _, name := range args { // This loop will fetch all the addons mentioned in the command and will remove them one by one.
```
Comment (shanshanying) [inline pkg/cmd/addon/addon.go:272]: pls add some unit tests for multiple args.

**Comment by SuperAayush** [comment]
I have been trying to add a test for the disable command part. Have tried to debug using multiple cases but all of them throw the same error - `The connection to the server localhost:8080 was refused - did you specify the right host or port?`. I understand the error is saying that my code addition is trying to connect to a K8s existing cluster but it throws the same error even if I make a fake K8s client.

Here are the 3 different code changes I done but got the same error:

1. https://gist.github.com/SuperAayush/c89a011d6e8c8acb6eeb1ff50818dc0b#file-gistfile1-txt-L105

2. https://gist.github.com/SuperAayush/7ba7838f6dec8138396728070dc47e35

3. https://gist.github.com/SuperAayush/7f6f7d63ba1b4f4f25b4d2e8e0c17c7c

I did also try to use `clientfake` instead of `restfake` but still no change in the error.

cc - @shanshanying

**Comment by shanshanying** [comment]
> Reference
The error is raised at line
https://github.com/superaayush/kbcli/blob/improvement/kbcli-disable-addons-in-batch-issue237/pkg/cmd/addon/addon.go#L303

When addon  cmd calls `ServerVersion()`. 
To solve this issue, you can refer to how `client-go` mock the test for this function at:
https://github.com/kubernetes/client-go/blob/1518fca9f06c6a73fc091535b8966c71704e657b/discovery/discovery_client_test.go#L48

Hope it will help.

**Comment by shanshanying** [comment]
resolve conflicts pls.

File: pkg/cmd/addon/addon_test.go [hunk 3]
```diff
@@ -22,21 +22,20 @@ package addon
...
 	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/dynamic/fake"
 	"k8s.io/client-go/kubernetes/scheme"
 	restfake "k8s.io/client-go/rest/fake"
 	cmdtesting "k8s.io/kubectl/pkg/cmd/testing"
-
-	"github.com/apecloud/kbcli/pkg/action"
-	"github.com/apecloud/kbcli/pkg/testing"
-	"github.com/apecloud/kbcli/pkg/types"
```
Comment (ldming) [inline pkg/cmd/addon/addon_test.go:39]: This import order does not need to be adjusted.

**Comment by SuperAayush** [comment]
Strange, I did not close the PR. No idea what happened here 🤔

**Comment by SuperAayush** [comment]
Strange, I did not close the PR. No idea what happened here 🤔

**Comment by SuperAayush** [comment]
Going for a new PR

**Comment by SuperAayush** [comment]
Please head over to this PR: https://github.com/apecloud/kbcli/pull/313

**Pull request closed without merging**